File: 9-palindrome-number/9-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        std::string str = std::to_string(x); 
        int a_ptr = 0, b_ptr = str.size()-1;
        
        while(a_ptr < b_ptr) {
            if(str[a_ptr] != str[b_ptr]) return false;
            a_ptr++;
            b_ptr--;
        }
        
        return true;
    }
};

File: 9-palindrome-number/9-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        c = str(x)
        a, b = 0, len(c)-1
        
        while a <= b:
            if c[a]!=c[b]:
                return False
            else:
                a = a+1
                b = b-1
        return True